File: z-clients/python/diom/models/msg_fifo_configure_in.py
# this file is @generated
import typing as t

from ..internal.base_model import BaseModel


class MsgFifoConfigureIn(BaseModel):
    namespace: str | None = None

    retry_schedule: t.List[int] | None = None

    dlq_topic: str | None = None


class _MsgFifoConfigureIn(BaseModel):
    namespace: str | None = None

    topic: str

    consumer_group: str

    retry_schedule: t.List[int] | None = None

    dlq_topic: str | None = None

File: z-clients/javascript/src/models/msgFifoConfigureIn.ts
// this file is @generated

export interface MsgFifoConfigureIn {
    namespace?: string | null;
    retrySchedule?: number[];
    dlqTopic?: string | null;
}

export interface MsgFifoConfigureIn_ {
    namespace?: string | null;
    topic: string;
    consumerGroup: string;
    retrySchedule?: number[];
    dlqTopic?: string | null;
}

export const MsgFifoConfigureInSerializer = {
    // biome-ignore lint/suspicious/noExplicitAny: intentional any
    _fromJsonObject(object: any): MsgFifoConfigureIn_ {
        return {
            namespace: object['namespace'],
            topic: object['topic'],
            consumerGroup: object['consumer_group'],
            retrySchedule: object['retry_schedule'],
            dlqTopic: object['dlq_topic'],
        };
    },

    // biome-ignore lint/suspicious/noExplicitAny: intentional any
    _toJsonObject(self: MsgFifoConfigureIn_): any {
        return {
            'namespace': self.namespace,
            'topic': self.topic,
            'consumer_group': self.consumerGroup,
            'retry_schedule': self.retrySchedule,
            'dlq_topic': self.dlqTopic,
        };
    }
}